typescript/sdk/src/hook/updates.ts

```diff
@@ -0,0 +1,184 @@
+import { Logger } from 'pino';
+import { Abi, encodeFunctionData } from 'viem';
+
+import {
+  Address,
+  EvmChainId,
+  deepCopy,
+  eqAddress,
+  isZeroishAddress,
+} from '@hyperlane-xyz/utils';
+
+import { CCIPContractCache } from '../ccip/utils.js';
+import { ContractVerifier } from '../deploy/verify/ContractVerifier.js';
+import { MultiProvider } from '../providers/MultiProvider.js';
+import { AnnotatedEV5Transaction } from '../providers/ProviderType.js';
+import { HypTokenRouterConfig } from '../token/types.js';
+import { ChainName } from '../types.js';
+import { extractIsmAndHookFactoryAddresses } from '../utils/ism.js';
+
+import { EvmHookModule } from './EvmHookModule.js';
+import { DerivedHookConfig } from './types.js';
+
+type ReadOnlyDerivedHookConfig = Readonly<DerivedHookConfig>;
+type ReadOnlyHookConfig = Readonly<NonNullable<HypTokenRouterConfig['hook']>>;
+
+type UpdateHookParams = {
+  evmChainId: EvmChainId;
+  evmChainName: ChainName;
+  mailbox: string;
+  proxyAdminAddress: string;
+  expectedConfig: ReadOnlyHookConfig;
+  actualConfig: ReadOnlyDerivedHookConfig | string;
+  logger: Logger;
+  hookAndIsmFactories: ReturnType<typeof extractIsmAndHookFactoryAddresses>;
+  multiProvider: MultiProvider;
+  setHookFunctionName: string;
+  contractToCallAbi: Abi;
+  ccipContractCache?: CCIPContractCache;
+  contractVerifier?: ContractVerifier;
+};
+
+export async function getEvmHookUpdateTransactions(
+  clientContractAddress: string,
+  updateHookParams: UpdateHookParams,
+): Promise<AnnotatedEV5Transaction[]> {
+  const updateTransactions: AnnotatedEV5Transaction[] = [];
+
+  const { expectedConfig: expectedHookConfig, actualConfig: actualHookConfig } =
+    updateHookParams;
+
+  if (
+    typeof expectedHookConfig === 'string' &&
+    isZeroishAddress(expectedHookConfig)
+  ) {
+    return [];
+  }
+
+  // Try to deploy or update Hook with the expected config
+  const {
+    deployedHook: expectedDeployedHook,
+    updateTransactions: hookUpdateTransactions,
+  } = await deployOrUpdateHook(updateHookParams);
```

Comment:
why not move the `deployOrUpdate` logic here into `update()` of the hook module more natively?

Reply:
that is a good point, I'll try and see what happens